lib/ansible/modules/cloud/azure/azure_rm_image_facts.py

```diff
@@ -0,0 +1,232 @@
+
+options:
+    resource_group:
+        description:
+            - Name of resource group.
+    name:
+        description:
+            - Name of the image to filter from existing images.
+    tags:
+        description:
+            - List of tags to be matched.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Madhura Naniwadekar(@Madhura-CSI)"
+'''
+
+
+EXAMPLES = '''
+- name: List images with name
+  azure_rm_image_facts:
+    name: test-image
+    resource_group: test-resource-group
+
+- name: List images by resource group
+  azure_rm_image_facts:
+    resource_group: test-resource-group
+    tags:
+      - testing
+      - foo:bar
+
+- name: List all available images under current subscription
+  azure_rm_image_facts:
+'''
+
+
+RETURN = '''
+azure_images:
+    description: List of image dicts.
+    returned: always
+    type: list
+    example: [{
+        "data_disks": [
+            {
+                "caching": "None",
+                "disk_size_gb": 50,
+                "lun": 0,
+                "managed_disk": {
+                    "id": "/subscriptions/xyz/resourceGroups/test-resource-group/providers/Microsoft.Compute/disks/disk1"
+                },
+                "storage_account_type": "Standard_LRS"
+            },
+            {
+                 "caching": "None",
+                 "disk_size_gb": 50,
+                 "lun": 1,
+                 "managed_disk": {
+                     "id": "/subscriptions/xyz/resourceGroups/test-resource-group/providers/Microsoft.Compute/disks/disk2"
+                 },
+                 "storage_account_type": "Standard_LRS"
+            }
+        ],
+        "location": "centralindia",
+        "name": "test-image",
+        "os_disk": "/subscriptions/xyz/resourceGroups/test-resource-group/providers/Microsoft.Compute/disks/testinstimg_OsDisk_1_XXXX",
+        "os_disk_caching": "ReadWrite",
+        "os_state": "Generalized",
+        "os_storage_account_type": "Standard_LRS",
+        "os_type": "Linux",
+        "provisioning_state": "Succeeded",
+        "resource_group": "test-resource-group",
+        "source": "/subscriptions/xyz/resourceGroups/test-resource-group/providers/Microsoft.Compute/virtualMachines/testinstimg",
+        "tags": {
+            "foo": "bar",
+            "type": "testimg"
+         }
+    }]
+'''
+
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+
+AZURE_ENUM_MODULES = ['azure.mgmt.compute.models']
+
+
+class AzureRMImageFacts(AzureRMModuleBase):
+
+    def __init__(self, **kwargs):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str'),
+            name=dict(type='str'),
+            tags=dict(type='list')
+        )
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_images=[])
```

Comment:
pls remove ansible_facts, and azure from the  result, just `images=[]`. we're working on general code guideline on facts module, one of them is shorten the result level, rule is no ansible_facts , and azure key word, since it's already  azure module.

Reply:
Please check the latest commit. I have removed 'azure' from resultant list name and removed the same from ansible_facts as well.